### PR TITLE
Iotedge 741 create Google IoT Core integration example

### PR DESCRIPTION
**Pull request by LimaEchoCharlie**
Google IoT Core integration example with a step by step guide. 
I will add a in-depth introduction in a following PR.

### CODE REVIEW

File: integrations/gcpiot/examples/src/forgerock.org/cmd/gcp-device-mqtt/main.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+	}{string(keyBytes)}
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		return
+	}
+
+	return string(dataBytes), nil
+}
+
+// createJWT returns a signed JWT that can be used to as a password for the MQTT server
```
Comment (JacoJooste) [inline integrations/gcpiot/examples/src/forgerock.org/cmd/gcp-device-mqtt/main.go]:
```suggestion
// createJWT returns a signed JWT that can be used as a password for the MQTT server
```

File: integrations/gcpiot/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+## Google Cloud Platform IoT Integration
+### Introduction
```
Comment (JacoJooste) [inline integrations/gcpiot/README.md]: Were you planning on adding an introduction or should this be removed?

Comment (LimaEchoCharlie) [reply integrations/gcpiot/README.md]: Yes, I am going to add an introduction next, this was just getting a bit on the large side.

File: integrations/gcpiot/am/groovy/CustomDeviceAttestationGCPDevice.groovy [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+String projectID = ""
+String region = ""
+String registryID = ""
+String serviceAccountCredentials = ""
+/*
+ *    End project data
+ */
+
+
+class retryHttpInitializerWrapper implements HttpRequestInitializer {
```
[the remark this answers is not in the document]
Comment (LimaEchoCharlie) [reply integrations/gcpiot/am/groovy/CustomDeviceAttestationGCPDevice.groovy]: Can you spot that I am a Gopher?

**Pull request merged**